java/memory/src/main/java/io/netty/buffer/ArrowBuf.java

```diff
@@ -86,17 +85,14 @@
    *
    * @param referenceManager The memory manager to track memory usage and reference count of this buffer
    * @param length The  byte length of this buffer
-   * @param isEmpty  Indicates if this buffer is empty which enables some optimizations.
    */
   public ArrowBuf(
       final ReferenceManager referenceManager,
       final BufferManager bufferManager,
       final long length,
-      final long memoryAddress,
-      boolean isEmpty) {
+      final long memoryAddress) {
```

Comment:
I am not sure if directly changing a public API would cause client code break.

Reply:
Agree we can't easily change the public api, but i think it's ok if it‘s really needed, and users could correct the behavior when they bump version.